nucypher/blockchain/eth/actors.py

```diff
@@ -109,24 +112,30 @@ class ActorError(Exception):
 
     @validate_checksum_address
     def __init__(self,
+                 domain: Optional[str],
                  registry: BaseContractRegistry,
-                 domain: Optional[str] = None,
+                 transacting_power: Optional[TransactingPower] = None,
                  checksum_address: Optional[ChecksumAddress] = None):
 
-        # TODO: Consider this pattern - None for address?.  #1507
-        # Note: If the base class implements multiple inheritance and already has a checksum address...
+        if not bool(checksum_address) ^ bool(transacting_power):
```

Comment:
Can you clarify operator precedence here? At first sight, I wouldn't know if this is `(not A) xor B` or `not (A xor B)`.